docs/spec/src/integration/spec/4-contracts.md

```diff
@@ -75,4 +76,23 @@ Contains EigenDA network registered Dispersers’ Ethereum address. The EigenDA
 
 ### EigenDACertVerifier
 
-This contract's main use case is exposing a function verifyDACertV2 which is used to verify `DACert`s. This function’s logic is described in the [Cert Validation](./6-secure-integration.md#cert-validation) section.
+This contract's main use case is exposing a function checkDACert which is used to verify `DACerts`. This function’s logic is described in the [Cert Validation](./6-secure-integration.md#cert-validation) section. 
+
+The contract also exposes a `certVersion` method which is called by the payload disperser client to understand which certificate version is needed for backend encoding. 
+
+### EigenDACertVerifierRouter
+
+This contract's main use case is allowing for secure upgrades of EigenDACertVerifier contracts in a cross version compatible format. This is done through maintaining a stateful mapping:
```

Comment:
we should prob mention that its not only for logic upgrades, but most importantly for upgrading custom quorums and security thresholds